src/QueryConditionGenerator.php

```diff
@@ -81,40 +105,79 @@ private function processGroup(ValuesGroup $group): array
+                    foreach ($values as $idx => $value) {
+                        $valuesBag->removeSimpleValue($idx);
+                        $value = $dateTransformer->transform($value);
+                        $range = new Range($value, $value);
+                        $valuesBag->add($range);
+                    }
+                } else {
+                    $bool[$includingType][]['terms'] = [$propertyName => array_values($valuesBag->getSimpleValues())];
+                }
             }
 
             if ($valuesBag->hasExcludedSimpleValues()) {
-                $bool['must_not'][]['terms'] = [$this->mappings[$fieldName]->indexName => array_values($valuesBag->getExcludedSimpleValues())];
+                $bool['must_not'][]['terms'] = [$propertyName => array_values($valuesBag->getExcludedSimpleValues())];
             }
 
-            /** @var Range $range */
-            foreach ($valuesBag->get(Range::class) as $range) {
-                $rangeParams = [
-                    $range->isLowerInclusive() ? 'lte' : 'lt' => $range->getLower(),
-                    $range->isUpperInclusive() ? 'gte' : 'gt' => $range->getUpper(),
-                ];
+            $isId = self::PROPERTY_ID === $propertyName;
 
-                $bool[$includingType][] = [$this->mappings[$fieldName]->indexName => $rangeParams];
-            }
+            if (false === $isId) {
+                /** @var Range $range */
+                foreach ($valuesBag->get(Range::class) as $range) {
+                    $rangeParams = [
+                        $range->isLowerInclusive() ? 'lte' : 'lt' => $range->getLower(),
+                        $range->isUpperInclusive() ? 'gte' : 'gt' => $range->getUpper(),
+                    ];
+
+                    $bool[$includingType][]['range'][$propertyName] = $rangeParams;
+                }
 
-            foreach ($valuesBag->get(ExcludedRange::class) as $range) {
-                $rangeParams = [
-                    $range->isLowerInclusive() ? 'lte' : 'lt' => $range->getLower(),
-                    $range->isUpperInclusive() ? 'gte' : 'gt' => $range->getUpper(),
-                ];
+                foreach ($valuesBag->get(ExcludedRange::class) as $range) {
+                    $rangeParams = [
+                        $range->isLowerInclusive() ? 'lte' : 'lt' => $range->getLower(),
+                        $range->isUpperInclusive() ? 'gte' : 'gt' => $range->getUpper(),
+                    ];
 
-                $bool['must_not'][] = [$this->mappings[$fieldName]->indexName => $rangeParams];
+                    $bool['must_not'][]['range'][$propertyName] = $rangeParams;
+                }
+            } else {
+                $ids = [];
+                foreach ($valuesBag->get(Range::class) as $range) {
+                    $ids = array_merge($ids, range($range->getLower(), $range->getUpper()));
+                }
+                if (false === empty($ids)) {
```

Comment:
No need for the explicit Boolean check, you can simple use `if (f!empty($ids)) {`
Same goes for all other code occurrences.